api/v1alpha1/component_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Kubebb Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Maintainer describes a Chart maintainer.
+type Maintainer struct {
+	// Name is a user name or organization name
+	Name string `json:"name,omitempty"`
+	// Email is an optional email address to contact the named maintainer
+	Email string `json:"email,omitempty"`
+	// URL is an optional URL to an address for the named maintainer
+	URL string `json:"url,omitempty"`
+}
+
+// ComponentSpec defines the desired state of Component
+type ComponentSpec struct {
+	// RepositoryRef is a reference to the Repository
+	RepositoryRef *corev1.ObjectReference `json:"repository"`
+	// The name of the component may come from helm chart name
+	Name string `json:"name"`
+	// versions contains all version of one component.
+	Versions []ComponetVersions `json:"versions"`
+	// FIXME: some fields(like description) may change when version update, how to deal with it?
+	// A one-sentence description of the chart
+	Description string `json:"description,omitempty"`
+	// Maintainers is a list of maintainers
+	Maintainers []string `json:"maintainers,omitempty"`
```

Comment:
Maybe.
```go
Maintainers []Maintainer `json:"maintainers,omitempty"`
```

Reply:
Good point. maintaniner can be a `name` `email` or even `blockchain address` `blockchain ens name`...

Reply:
helm chart use this struct to track maintainer.
```go
// Maintainer describes a Chart maintainer.
type Maintainer struct {
	// Name is a user name or organization name
	Name string `json:"name,omitempty"`
	// Email is an optional email address to contact the named maintainer
	Email string `json:"email,omitempty"`
	// URL is an optional URL to an address for the named maintainer
	URL string `json:"url,omitempty"`
}
```

Reply:
good enough! We can reuse `name` or `url` for blockchain address